metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -373,6 +373,33 @@ def read_file_spark(self, file: str, ext: str) -> Optional[DataFrame]:
         # see https://mungingdata.com/pyspark/avoid-dots-periods-column-names/
         return df.toDF(*(c.replace(".", "_") for c in df.columns))
 
+
+    def infer_file_without_extension(self, file):
+        fields = []
+        try:
+            fields = parquet.ParquetInferrer().infer_schema(file)
```

Comment:
What happens if none of the inferrer works? There's a chance that this could happen.

minor: A cleaner way without the nested try/excepts could be something like:
```python
for inferrer in (
    parquet.ParquetInferrer(),
    json.JsonInferrer(),
    csv_tsv.CsvInferrer(max_rows=self.source_config.max_rows),
    csv_tsv.TsvInferrer(max_rows=self.source_config.max_rows),
):
    file.seek(0)
    try:
        return inferrer.infer_schema(file)
    except:
        pass
# im not sure if file.filename exists
raise RuntimeError(f'Failed to infer schema for {file.filename}')
```

Reply:
:+1: I see, if we want to catch certain exceptions and let the rest bubble up, then yeah we'd need to be more explicit and won't work in the case i suggested.